File: Leetcode/1217.cpp
class Solution {
public:
    int getCost(int src, int dest) {
        return ((dest - src) % 2 == 0) ? 0 : 1;
    }
    int minCostToMoveChips(vector<int>& position) {
        unordered_set<int> pos(position.begin(), position.end());
        int minCost = numeric_limits<int>::max();
        int currentCost;
        for (const auto &i : pos) {
            currentCost = 0;
            for (const auto &j : position) {
                currentCost += getCost(j, i);
            }
            minCost = min(minCost, currentCost);
        }
        return minCost;
    }
};

File: Leetcode/1217/1217.go
func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

func solve(from, to int) int {
	if from == to {
		return 0
	}
	if abs(from - to) % 2 == 0 {
		return 0
	}
	return 1
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func minCostToMoveChips(position []int) int {
	minCost := math.MaxInt32
	for i := 0; i < len(position); i++ {
		sum := 0
		for j := 0; j < len(position); j++ {
			sum += solve(position[j], position[i])
		}
		minCost = min(minCost, sum)
	}
	return minCost
}